src/Microsoft.AspNetCore.Server.Kestrel.Core/Features/IHttpRequestBodyTimeoutFeature.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Features
+{
+    /// <summary>
+    /// Represents a timeout for the request body in an HTTP request.
+    /// </summary>
+    public interface IHttpRequestBodyTimeoutFeature
+    {
+        /// <summary>
+        /// The minimum amount of time allowed for the request body to be entirely read.
+        /// </summary>
+        TimeSpan Timeout { get; }
+
+        /// <summary>
+        /// The maximum amount of time allowed for the request body to be entirely read.
+        /// Only effective when <see cref="MinimumDataRate"/> is also set.
+        /// </summary>
+        TimeSpan? ExtendedTimeout { get; }
```

Comment:
Still haven't resolved the redundancy between ExtendedTimeout and MinimumDataRate. MinimumDataRate scales with the size of the body, ExtendedTimeout doesn't. ExtendedTimeout will be a pain for any customer that tries to use it.